docs/setup/setup.md

```diff
@@ -17,110 +23,127 @@
-
-### Clone KubeEdge
+## Run KubeEdge
+
+### Run on cloud side
+
+#### Clone KubeEdge
 
 ```shell
 git clone https://github.com/kubeedge/kubeedge.git $GOPATH/src/github.com/kubeedge/kubeedge
 cd $GOPATH/src/github.com/kubeedge/kubeedge
 ```
 
-### Configuring MQTT mode
-
-The Edge part of KubeEdge uses MQTT for communication between deviceTwin and devices. KubeEdge supports 3 MQTT modes:
-1) internalMqttMode: internal mqtt broker is enabled.
-2) bothMqttMode: internal as well as external broker are enabled.
-3) externalMqttMode: only external broker is enabled.
-
-Use mode field in [edge.yaml](https://github.com/kubeedge/kubeedge/blob/master/edge/conf/edge.yaml#L4) to select the desired mode.
-
-To use KubeEdge in double mqtt or external mode, you need to make sure that [mosquitto](https://mosquitto.org/) or [emqx edge](https://www.emqx.io/downloads/edge) is installed on the edge node as an MQTT Broker.
-
-### Generate Certificates
+#### Generate Certificates
 
 RootCA certificate and a cert/key pair is required to have a setup for KubeEdge. Same cert/key pair can be used in both cloud and edge.
 
 ```bash
 # $GOPATH/src/github.com/kubeedge/kubeedge/build/tools/certgen.sh genCertAndKey edge
 ```
 
-The cert/key will be generated in the `/etc/kubeedge/ca` and `/etc/kubeedge/certs` respectively.
-
-## Run KubeEdge
-
-### Run Cloud
+The cert/key will be generated in the `/etc/kubeedge/ca` and `/etc/kubeedge/certs` respectively. We need to copy these files to the corresponding edge side server directory.
 
 #### Run as a binary
 
 + Firstly, make sure gcc is already installed on your host. You can verify it via:
-    ```shell
-    gcc --version
-    ```
-
-+ Build Cloud and edge
 
     ```shell
-    cd $GOPATH/src/github.com/kubeedge/kubeedge
-    make 
+    gcc --version
     ```
 
-+ Build Cloud
++ Build cloudcore 
 
     ```shell
-    cd $GOPATH/src/github.com/kubeedge/kubeedge
-    make all WHAT=cloud
+    cd $GOPATH/src/github.com/kubeedge/kubeedge/
+    make all WHAT=cloud 
     ```
-
-+ The path to the generated certificates should be updated in `$GOPATH/src/github.com/kubeedge/kubeedge/cloud/conf/controller.yaml`. Please update the correct paths for the following :
-    + cloudhub.ca
-    + cloudhub.cert
-    + cloudhub.key
-
 + Create device model and device CRDs.
+
     ```shell
     cd $GOPATH/src/github.com/kubeedge/kubeedge/build/crds/devices
     kubectl create -f devices_v1alpha1_devicemodel.yaml
     kubectl create -f devices_v1alpha1_device.yaml
     ```
 
-+ Run cloud
++ Copy cloudcore binary and config file 
+
     ```shell
     cd $GOPATH/src/github.com/kubeedge/kubeedge/cloud
     # run edge controller
     # `conf/` should be in the same directory as the cloned KubeEdge repository
     # verify the configurations before running cloud(cloudcore)
-    ./cloudcore
+    mkdir ~/cmd/conf
+    cp cloudcore ~/cmd/
+    cp -rf conf/* ~/cmd/conf/
```

Comment:
hard code here? User maybe can move to anywhere they want.

Reply:
this is an example, we can comment it